app/controllers/file_controller.rb

```diff
@@ -50,7 +50,9 @@ def show
     disposition = params[:disposition] || 'inline'
     filename = params["filename"] || "#{objid}#{fileid}"
     headers['Content-Disposition'] = "#{disposition}; filename=#{filename}"
-    if ds.mimeType
+    if filename.include?('.jpg')
```

Comment:
@hweng I am assuming any jpeg derivatives we generated in the past that don't have a correct mimetype is corrupted in some extend, and we had better re-create them. Have you got a list of all jpeg derivatives that don't have a correct mimetype?